src/Field/Image.php

```diff
@@ -116,6 +122,10 @@ protected function fillThumbnailFields(array $data)
         $size->height = $data['height'];
         $size->mime_type = $data['mime-type'];
 
+        $imgDir = substr($this->url, 0, strrpos($this->url, '/'));
+        $size->url = $imgDir.'/'.$data['file'];
```

Comment:
This is not being tested. If you can please write a test. I guess no fields are being tested, so if you can please test all of them.